src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+  const result = [];
+
+  actions.map(({ type, keysToRemove, extraData }) => {
+    switch (type) {
+      case 'clear':
+        Object.keys(copyState).map(key => delete copyState[key]);
+        break;
+
+      case 'removeProperties':
+        keysToRemove.map(key => copyState[key] && delete copyState[key]);
```

Comment:
```suggestion
        keysToRemove.forEach(key => copyState[key] && delete copyState[key]);
```

Reply:
@OlEzhA0 Can you argue why forEach and not map?